lib/components/beta-bar/index.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import SmallCrossIcon from '../../icons/cross-small';
+import { isElectron } from '../../utils/platform';
+
+const BetaBar = () => {
+  const [isVisible, setIsVisible] = useState(true);
+
+  const shouldShowBanner = () => {
+    if (isElectron) {
+      return false;
+    }
+    try {
+      const stored = localStorage.getItem('betaBannerDismissedAt');
+      const dismissedAt = stored ? parseInt(stored, 10) : -Infinity;
+
+      return (
+        isNaN(dismissedAt) || Date.now() - dismissedAt > 7 * 24 * 60 * 60 * 1000
+      );
+    } catch (e) {
+      return true;
+    }
+  };
+
+  const dismissBanner = () => {
+    setIsVisible(false);
+    try {
+      localStorage.setItem('betaBannerDismissedAt', Date.now().toString());
+    } catch (e) {
+      // pass
+    }
+  };
```

Comment:
@belcherj is right about not wanting to call this on every render. we can put it in `useEffect()` or more ideally use it to set the default value for `useState()` since we never want to make the banner appear once we've loaded if it shouldn't have already been showing at the start of loading

```ts
const loadDismissal = (): boolean => {
	try {
		const stored = localStorage.getItem('betaBannerDismissedAt');
		const dismissedAt = stored ? parseInt( stored, 10 ) : -Infinity;

		return isNaN( dismissedAt ) || ( Date.now() - dismissedAt ) > 7 * 24 * 60 * 60 * 1000;
	} catch ( e ) {
		return true;
	}
}

const storeDismissal = () => {
	try {
		localStorage.setItem( 'betaBannerDismissedAt', Date.now() );
	} catch ( e ) {
		// pass
	}
}

const BetaBar = () => {
	const [isVisible, setIsVisible] = useState(loadDismissal());
	const dismissBanner = () => { storeDismissal(); setIsVisible(false); };

	…
}
```

Reply:
this is very clever thank you!